dpgv4.py

```diff
@@ -551,11 +553,24 @@ def convert_file(input_file: str, output_file: str, options: Any) -> None:
     )
     v_tmp_file = TemporaryFile()
     encode_stream("video", v_cmd, v_tmp_file)
-    a_cmd = prepare_audio_conversion_command(input_file, options.aid)
-    a_tmp_file = TemporaryFile()
-    encode_stream("audio", a_cmd, a_tmp_file)
+    #Create blank audio if no codec exists
+    try:
+        a_cmd = prepare_audio_conversion_command(input_file, options.aid)
+        a_tmp_file = TemporaryFile()
+        encode_stream("audio", a_cmd, a_tmp_file)
+    except ExternalCommandFailedError:
+        a_cmd = prepare_audio_conversion_command(input_file, options.aid)
+        a_tmp_file = TemporaryFile()
+        for i in list(reversed(['-t',str(count_video_frames(v_tmp_file) / options.framerate),'-f','lavfi','-i','anullsrc'])):
+            a_cmd.insert(1,i)
+        encode_stream("audio", a_cmd, a_tmp_file)
     gop = create_gop(v_tmp_file)
-    thumbnail = create_thumbnail(create_screenshot(v_tmp_file, int(get_duration(input_file) / 10)))
+    #Catch bad thumbnails
+    try:
+        thumbnail = create_thumbnail(create_screenshot(v_tmp_file, int(get_duration(input_file) / 10)))
+    except:
+        print("Bad Thumbnail")
+        exit()
```

Comment:
1) This would not only catch "bad thumbnails" (whatever that means), but also errors originating from the `create_screenshot` and `get_duration` functions. For example, if the ffmpeg duration command crashed for some reason, we'd get a misleading "Bad Thumbnail" message. This would only make it harder to find the real reason for the failure.

2) Why catch the error at all if we're not doing anything about it? As I understand it, we'd get a short "Bad Thumbnail" message instead of an exception traceback. This is useless, because this short message doesn't help with debugging, while the exception does.

3) The message would be printed on the [stdout](https://en.wikipedia.org/wiki/Standard\_streams#Standard\_output_(stdout)) instead of stderr, which is bad practice for error messages.

4) The program's exit code in case of caught bad thumbnail would change to 0, which indicates successfull completion, while the program in fact did not successfully complete its task.

To sum up: this change is not going to be applied.

If you have a need to handle errors during thumbnail generation, please explain how would you like the error to be handled. We could use some kind of a placeholder thumbnail, but handling errors via hiding them is not acceptable.